File: homework/home4.1/src/CustomPizzaBuilder.java
import java.util.ArrayList;
import java.util.List;

public class CustomPizzaBuilder implements PizzaBuilder {
    private List<Ingredient> selectedIngredients = new ArrayList<>();

    @Override
    public void addIngredient(Ingredient ingredient) {
        selectedIngredients.add(ingredient);
    }

    @Override
    public Pizza buildPizza() {
        return new Pizza(new ArrayList<>(selectedIngredients));
    }
}

File: classwork/classwork4.1/CustomPizzaBuilder.cpp
#pragma once


#include "PizzaBuilder.cpp"

// Concrete implementation of PizzaBuilder for creating custom pizzas
class CustomPizzaBuilder : public PizzaBuilder {
public:
    std::unique_ptr<Pizza> buildPizza() override {
        return std::make_unique<Pizza>(selectedIngredients);
    }
};
